papers/Sepideh_Alassi_Bertram_Winter/PyCells_for_an_Open_Semiconductor_Industry

```diff
@@ -0,0 +1,270 @@
+
+
+Development
+-----------
+
+OpenAceess [OpenAccess]_ has been established as standard for storing design data in the semiconductor industry and builds the foundation for interoperability of EDA-tools. OpenAccess is a file based database and manages logical design data (schematics) as well as artwork data (layout) for manufacturing. The reference implementation for accessing design data within the database is written in C++. The company Ciranova (now acquired by Synopsys) developed a Python wrapper [PythonAPI]_ for the OpenAccess C++ class library with the goal to access and modify design data through Python. Additionally, they provide an integrated development environment [SynopsysPyCellStudio]_ that enables interactive development and debugging of Python code, while the effects on the design data are shown directly in a separate window of the GUI.
+
+In our development, we have used Python API library to write Python codes which generate recurring layout structures based on a given set of parameters. Typical recurring structures are primitive devices that are used in integrated circuit design such as resistors, capacitors and transistors. The electrical properties of the mentioned devices are brought down to geometric dimensions (e. g. width and length) which serve as parameters for the Python scripts. The object oriented nature of Python matches very well with this task as shown in the following example:
+
+A typical CMOS semiconductor process offers n-channel and p-channel transistors. Each of these transistor types can have multiple voltage ratings (e. g. 5V, 20V, 50V 120V), accordingly resulting in different layout structures. In general, transistors have similar gate, source and drain structures, yet they have eventually different doping concentrations for the implant regions of the semiconductor material and different protection structures, which need to be reflected in the layout. Therefore, we have created a set of base classes in Python from which the more specific transistors inherit common structures, while adding only structures dedicated to the type of transistor. This approach reduces duplicate code and makes maintaining the scripts more intuitive when the number of devices increases. Finally the Python scripts are compiled to byte-code and stored inside the OpenAccess library. A plugin of the OpenAccess database ascertains loading the correct script when a parameter of the corresponding device is modified. In the next section, the development of PyCells is explained using a sample high voltage transistor. 
```

Comment:
I do not think 'eventually' is the word you are looking for in this paragraph. Perhaps you mean 'extremely' different?